lib/ignored-paths.js

```diff
@@ -157,6 +174,25 @@ class IgnoredPaths {
                 this.baseDir = path.dirname(path.resolve(options.cwd, ignorePath));
                 addIgnoreFile(this.ig.custom, ignorePath);
                 addIgnoreFile(this.ig.default, ignorePath);
+            } else {
+                try {
+
+                    // if the ignoreFile does not exist, check package.json for eslintIgnore
+                    const packageJSONPath = findPackageJSONFile(options.cwd);
+
+                    if (packageJSONPath) {
+                        const packageJSONOptions = JSON.parse(fs.readFileSync(packageJSONPath, "utf8"));
+
+                        if (packageJSONOptions.eslintIgnore) {
+                            packageJSONOptions.eslintIgnore.forEach(pattern => {
```

Comment:
Can you add an assertion that `packageJSONOptions.eslintIgnore` is an array, rather than just checking that it exists? If someone accidentally has a file like this, then eslint will currently throw a confusing error.

```json
{
  "eslintIgnore": "foo.js"
}
```